library/proc_macro/src/quote.rs

```diff
@@ -46,11 +48,11 @@ macro_rules! quote_ts {
 ///
 /// Note: supported tokens are a subset of the real `quote!`, but
 /// unquoting is different: instead of `$x`, this uses `(@ expr)`.
-macro_rules! quote {
+macro_rules! minimal_quote {
     () => { TokenStream::new() };
     ($($t:tt)*) => {
         [
-            $(TokenStream::from(quote_ts!($t)),)*
+            $(ToTokens::into_token_stream(minimal_quote_ts!($t)),)*
         ].iter().cloned().collect::<TokenStream>()
```

Comment:
Similar to the above, not your code but can be cleaned up since arrays can now do `into_iter`.

```rust
        ].into_iter().collect::<TokenStream>()
```